libs/s25client/s25client.cpp

```diff
@@ -278,45 +280,95 @@ void SetAppSymbol()
+bool MigrateFilesAndDirectories()
 {
-    // Note: Do not use logger yet. Filepath may not exist
-    const std::string curPath = bfs::current_path().string();
-    LOG.write("Starting in %s\n", LogTarget::Stdout) % curPath;
-
-    // diverse dirs anlegen
-    std::array<unsigned, 9> dirs = {{94, 41, 47, 48, 51, 85, 98, 99, 100}}; // settingsdir muss zuerst angelegt werden (94)
-
-    std::string oldSettingsDir;
+    struct MigrationEntry
+    {
+        std::string oldName, newName;
+        bool isFolder;
+    };
+    const std::string sharedLibext =
+#ifdef _WIN32
+      "dll";
+#elif defined(__APPLE__)
+      "dylib";
+#else
+      "so";
+#endif
 
+    // Mapping from old files or directories to new ones
+    // Handled in order so multiple renamings are possible
+    // Empty newName = Delete
+    const std::vector<MigrationEntry> migrations = {
 #ifdef _WIN32
-    oldSettingsDir = RTTRCONFIG.ExpandPath("~/Siedler II.5 RttR");
+      {"~/Siedler II.5 RttR", s25::folders::config, true},
 #elif defined(__APPLE__)
-    oldSettingsDir = RTTRCONFIG.ExpandPath("~/.s25rttr");
+      {"~/.s25rttr", s25::folders::config, true},
 #endif
-    if(!oldSettingsDir.empty() && bfs::is_directory(oldSettingsDir))
+      {std::string(s25::folders::lstsUser).append("/SOUND.LST"), "", false},
+      {std::string(s25::folders::driver).append("/video/libvideoSDL.").append(sharedLibext), "", false},
+    };
+
+    try
     {
-        const std::string newSettingsDir = RTTRCONFIG.ExpandPath(FILE_PATHS[94]);
-        if(bfs::exists(newSettingsDir))
-        {
-            s25util::error(std::string("Old and new settings directory found. Please delete the one you don't want to keep!\nOld: ")
-                           + oldSettingsDir + "\nNew: " + newSettingsDir);
-            return false;
-        }
-        boost::system::error_code ec;
-        bfs::rename(oldSettingsDir, newSettingsDir, ec);
-        if(ec != boost::system::errc::success)
+        for(const MigrationEntry& entry : migrations)
         {
-            s25util::error(std::string("Old settings directory found at ") + oldSettingsDir
-                           + "\n Renaming to new name failed: " + newSettingsDir + "\nError: " + ec.message()
-                           + "\nRename it yourself and/or make sure the directory is writable!");
-            return false;
+            const bfs::path oldName = RTTRCONFIG.ExpandPath(entry.oldName);
+            if(!exists(oldName))
+                continue;
+            if(entry.isFolder && !is_directory(oldName))
+                throw std::runtime_error(helpers::format("Expected folder but %1% is not a folder", oldName));
+            else if(!entry.isFolder && !is_regular_file(oldName))
+                throw std::runtime_error(helpers::format("Expected file but %1% is not a file", oldName));
+            if(entry.newName.empty())
+            {
+                boost::system::error_code ec;
+                if(!remove(oldName, ec))
```

Comment:
perhaps we should be verbose here, that the user nows whats going on.

Reply:
https://github.com/Return-To-The-Roots/s25client/pull/1254/commits/ddc68f779ab096f1b7aff5b7501c28ac8d83fd09